data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/cnec/FlowCnecCreator.java

```diff
@@ -0,0 +1,321 @@
+package com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.cnec;
+
+import com.farao_community.farao.commons.Unit;
+import com.farao_community.farao.data.crac_api.Contingency;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_api.Instant;
+import com.farao_community.farao.data.crac_api.cnec.FlowCnecAdder;
+import com.farao_community.farao.data.crac_api.cnec.Side;
+import com.farao_community.farao.data.crac_api.threshold.BranchThresholdAdder;
+import com.farao_community.farao.data.crac_creation.creator.api.ImportStatus;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileConstants;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileCracCreationContext;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileElementaryCreationContext;
+import com.powsybl.iidm.network.*;
+import com.powsybl.triplestore.api.PropertyBag;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.*;
+
+public class FlowCnecCreator extends AbstractCnecCreator {
+
+    private enum FlowCnecDefinitionMode {
+        CONDUCTING_EQUIPMENT,
+        OPERATIONAL_LIMIT,
+        WRONG_DEFINITION;
+    }
+
+    private final String conductingEquipment;
+    private final Set<Side> defaultMonitoredSides;
+
+    public FlowCnecCreator(Crac crac, Network network, String assessedElementId, String nativeAssessedElementName, String assessedElementOperator, boolean inBaseCase, PropertyBag currentLimitPropertyBag, String conductingEquipment, List<Contingency> linkedContingencies, Set<CsaProfileElementaryCreationContext> csaProfileCnecCreationContexts, CsaProfileCracCreationContext cracCreationContext, Set<Side> defaultMonitoredSides, String rejectedLinksAssessedElementContingency) {
+        super(crac, network, assessedElementId, nativeAssessedElementName, assessedElementOperator, inBaseCase, currentLimitPropertyBag, linkedContingencies, csaProfileCnecCreationContexts, cracCreationContext, rejectedLinksAssessedElementContingency);
+        this.conductingEquipment = conductingEquipment;
+        this.defaultMonitoredSides = defaultMonitoredSides;
+    }
+
+    public void addFlowCnecs() {
```

Comment:
Beware of method names : there are addFlowCnecs, addFlowCnec, addFlowCnecData, addAllFlowCnecs in this class. I think you should find more explicit names, mainly for the last two.